### ISSUE TITLE
UpdateRecord requires Name field

### ISSUE THREAD

**Issue by Ulexus** [dnsimple/dnsimple-go#33]
The `Name` field of `ZoneRecord` does not have `omitempty` set on its tag, so it sends it even if it is empty, which causes the server to respond with `400 Validation failed`.

The simple workaround is to always supply the `Name` field in the `ZoneRecord`, but this does not agree with the API documentation.

This should either be documented with `UpdateRecord` or have `ZoneRecord` updated to make the `Name` field tagged with `omitempty`.

**Comment by weppos** [comment]
@Ulexus the `omitempty` was intentionally removed on https://github.com/aetrion/dnsimple-go/pull/16 because we require to provide a blank `""` string when you create a record for the apex domain.

I'm pretty sure it also worked for the updates at the time we tried, but indeed it seems to be a reasonable issue that I also pointed out at the time of #16.

We'll try to find a solution.

**Comment by Ulexus** [comment]
I, too, think it worked originally.  I had code out that suddenly stopped working at some point a few months ago, and I finally had time to track it down to this.

I don't think there is anything wrong with this compromise, I would just at least document it, since the requirement varies from the API spec.  For my case, I was already calling GetRecord() first, anyway, so I just added the Name field to the update... I just needed to _know_ that was required.
